src/modules/leaderboard.ts

```diff
@@ -0,0 +1,182 @@
+import * as Discord from "discord.js";
+import * as Builders from "@discordjs/builders";
+
+import { CommandDescriptor } from "../bot.d";
+
+import * as utils from "./utils";
+
+const MAX_PEOPLE = 50;
+
+type UsersCharacterCount = Discord.Collection<Discord.Snowflake, number>;
+
+export async function getUsersCharacterCount(
+	guild: Discord.Guild,
+	onlyCountAfter?: Date
+): Promise<[UsersCharacterCount, number, number, Discord.Channel[]]> {
+	// TODO: checkpoint
```

Comment:
Will this be included in this PR, or will it be added later?
I think the bot should cache the number of characters in messages older than a month and keep the timestamp. When doing it again, just go up to the timestamp and update the cache accordingly.

Reply:
1 month might interfere with mechanism to get leaderboard for last month; caching messages older than 45 days instead